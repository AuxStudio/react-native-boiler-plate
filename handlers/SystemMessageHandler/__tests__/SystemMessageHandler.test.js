import React from 'react';
import renderer from 'react-test-renderer';
import { View } from 'react-native';

import { SystemMessageHandler } from '..';

jest.mock('react-native-snackbar', () => {
  return {
    show: jest.fn(),
    hide: jest.fn(),
  };
});

describe('SystemMessageHandler', () => {
  let spy;
  const dispatch = jest.fn();

  it('renders with all props', () => {
    expect(
      renderer.create(
        <SystemMessageHandler
          dispatch={jest.fn()}
          systemMessage="Something went wrong"
          uid="xxxx-xxxx-xxxx-xxxx"
        >
          <View />
        </SystemMessageHandler>,
      ),
    ).toMatchSnapshot();
  });

  it('renders with minimum required props', () => {
    expect(
      renderer.create(
        <SystemMessageHandler dispatch={jest.fn()}>
          <View />
        </SystemMessageHandler>,
      ),
    ).toMatchSnapshot();
  });

  it('shows the snackbar if systemMessage changed in componentDidUpdate', () => {
    const component = renderer.create(
      <SystemMessageHandler dispatch={dispatch} systemMessage="Something went wrong">
        <View />
      </SystemMessageHandler>,
    );
    const instance = component.getInstance();
    spy = jest.spyOn(instance, 'showSnackbar');

    component.update(
      <SystemMessageHandler dispatch={dispatch} systemMessage="Same same, but different">
        <View />
      </SystemMessageHandler>,
    );

    expect(spy).toHaveBeenCalled();
  });

  it('shows the snackbar if systemMessage changed in componentDidUpdate and calls resetError after timeout', (done) => {
    const component = renderer.create(
      <SystemMessageHandler dispatch={dispatch}>
        <View />
      </SystemMessageHandler>,
    );
    const instance = component.getInstance();
    spy = jest.spyOn(instance, 'showSnackbar');

    component.update(
      <SystemMessageHandler dispatch={dispatch} systemMessage="Something went wrong">
        <View />
      </SystemMessageHandler>,
    );

    expect(spy).toHaveBeenCalled();

    spy.mockClear();

    spy = jest.spyOn(instance, 'resetError');

    setTimeout(() => {
      expect(spy).toHaveBeenCalled();
      expect(dispatch).toMatchSnapshot();

      done();
    }, instance.snackbarDuration);
  });

  afterEach(() => {
    if (spy) {
      spy.mockClear();
    }
  });
});
