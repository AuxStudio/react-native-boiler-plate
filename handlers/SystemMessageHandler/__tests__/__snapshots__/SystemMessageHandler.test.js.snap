// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SystemMessageHandler renders with all props 1`] = `<View />`;

exports[`SystemMessageHandler renders with minimum required props 1`] = `<View />`;

exports[`SystemMessageHandler shows the snackbar if systemMessage changed in componentDidUpdate and calls resetError after timeout 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "type": "RESET_SYSTEM_MESSAGE",
      },
    ],
    Array [
      Object {
        "type": "RESET_SYSTEM_MESSAGE",
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;
